tests/api/utils/test_scheduler.py

```diff
@@ -1155,6 +1157,39 @@ async def test_update_schedule_failure_not_found(db: Session, scheduler: Schedul
     assert "Schedule not found" in str(excinfo.value)
 
 
+@pytest.mark.asyncio
+async def test_update_schedule_failure_not_found_in_scheduler(
+    db: Session, scheduler: Scheduler
+):
+    schedule_name = "schedule-name"
+    project_name = config.default_project
+    scheduled_object = _create_mlrun_function_and_matching_scheduled_object(
+        db, project_name
+    )
+
+    # create the schedule only in the db
+    inactive_cron_trigger = schemas.ScheduleCronTrigger(year="1999")
+    get_db().create_schedule(
+        db,
+        project_name,
+        schedule_name,
+        schemas.ScheduleKinds.job,
+        scheduled_object,
+        inactive_cron_trigger,
+        1,
+    )
+
+    with pytest.raises(mlrun.errors.MLRunNotFoundError) as excinfo:
+        scheduler.update_schedule(
+            db, mlrun.api.schemas.AuthInfo(), project_name, schedule_name
+        )
+    job_id = scheduler._resolve_job_id(project_name, schedule_name)
+    assert (
+        f"Schedule job with id {job_id} not found in scheduler. Reload is required."
+        in str(excinfo.value)
+    )
```

Comment:
can you _reload_ and shows it works as expected?

Reply:
Currently, reload is done only by restarting the API chief (or service).
We need to add an endpoint to trigger reload manually  - we'll need to discuss it offline.
We had this issue on a live system and restarting the chief fixed it.